src/printer.js

```diff
@@ -626,7 +628,7 @@ function genericPrintNoParens(path, options, print) {
           concat([
             leftBrace,
             indent(
-              options.tabWidth,
+              options.tabWidth * (parentIsUnionTypeAnnotation ? 2 : 1),
```

Comment:
I think we need to calculate it differently. This is a special case where we are trying to line something up specifically. We want to indent it according to `| {`, which takes up 3 characters. So I think this should just be 

`options.tabWidth + (parentIsUnionTypeAnnotation ? 3 : 0)`

Because it doesn't matter if the tab width is 1 or 5; you always want to bump everything over 3 spaces because that's what you're trying to line up with.